runtime/src/blueoil.cpp

```diff
@@ -19,6 +19,9 @@
 namespace blueoil {
 
 int Tensor::shapeVolume() {
+    return this->shapeVolume(shape);
+}
+int Tensor::shapeVolume(std::vector<int> shape) {
```

Comment:
Should it still remain to be a member if it doesn't access tensor state?

Free standing helper could be better for the purpose.

Do we really need pass `shape` by value?

Reply:
```
just
std::vector<int> shape() const;
should be fine, as here is no way to change state of vector itself
```
yes.
https://github.com/blue-oil/blueoil/pull/9/commits/64066ac3917fb6bcd1b6fe8090f7d81246992cba

Reply:
1. What is wrong with `const std::vector<int>&` here?
2. What is wrong with making it just a standalone helper in private namespace like:

```cpp
namespace {

int shapeVolume(const std::vector<int>& shape) {
  return std::accumulate(...);
}

} // namespace

namespace blueoil {

int Tensor::shapeVolume() const {
  return shapeVolume(m_shape);
}

} // namespace blueoil
```

So we will not attach anything which does not access class state as a method.